### PR TITLE
pacific: mgr/autoscaler: Introduce noautoscale flag

### PR DESCRIPTION
**Pull request by kamoltat**
`noautoscale` flag is a feature where the
user can choose to flip the switch between
turning autoscale `on` and `off` for all
pools with a single command.

`osd pool set noautoscale` will turn all
autoscale mode`off` for all pools.

`osd pool unset noautoscale` will turn all
autoscale mode `on` for all pools.

Fixes: https://tracker.ceph.com/issues/53769

Backporting the relevant commits from master PR:
#43716

Signed-off-by: Kamoltat ksirivad@redhat.com

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [x] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [x] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by kamoltat** [comment]
jenkins rerun test make check

**Comment by kamoltat** [comment]
jenkins test make check

**Comment by kamoltat** [comment]
Failed: 1/1
https://pulpito.ceph.com/ksirivad-2022-01-12_15:01:03-rados:singleton:all:test-noautoscale-flag.yaml-wip-ksirivad-backport-pacific-43716-distro-basic-smithi/

```/home/ubuntu/cephtest/clone.client.0/qa/workunits/mon/test_noautoscale_flag.sh
2022-01-12T16:14:07.118 INFO:tasks.workunit.client.0.smithi124.stderr:+ unset CEPH_CLI_TEST_DUP_COMMAND
2022-01-12T16:14:07.119 INFO:tasks.workunit.client.0.smithi124.stderr:++ ceph osd pool ls
2022-01-12T16:14:07.119 INFO:tasks.workunit.client.0.smithi124.stderr:++ wc -l
2022-01-12T16:14:07.394 INFO:tasks.workunit.client.0.smithi124.stderr:+ NUM_POOLS=1
2022-01-12T16:14:07.394 INFO:tasks.workunit.client.0.smithi124.stderr:+ '[' 1 -gt 0 ']'
2022-01-12T16:14:07.395 INFO:tasks.workunit.client.0.smithi124.stdout:test requires no preexisting pools
2022-01-12T16:14:07.396 DEBUG:teuthology.orchestra.run:got remote process result: 1
2022-01-12T16:14:07.396 INFO:tasks.workunit.client.0.smithi124.stderr:+ echo 'test requires no preexisting pools'
2022-01-12T16:14:07.397 INFO:tasks.workunit.client.0.smithi124.stderr:+ exit 1
2022-01-12T16:14:07.397 INFO:tasks.workunit:Stopping ['mon/test_noautoscale_flag.sh'] on client.0...```

test did not delete device_health_metrics, need to modify the test to delete it.

**Comment by kamoltat** [comment]
Still need to be tested against the rados suite to make sure it's clean

File: PendingReleaseNotes [hunk 1]
```diff
@@ -35,6 +35,13 @@
 >=16.2.7
 --------
 
+* MGR: The pg_autoscaler can now be turned `on` and `off` globally
+  with the `noautoscale` flag. By default this flag is unset and
+  the default pg_autoscale mode remains the same.
+  For more details, see:
+
+  https://docs.ceph.com/en/latest/rados/operations/placement-groups/
```
Comment (neha-ojha) [inline PendingReleaseNotes:43]: nit: should be under 16.2.8 now

**Review by neha-ojha — APPROVED** [review]

**Review by vumrao — APPROVED** [review]

**Comment by neha-ojha** [comment]
jenkins test api

**Comment by neha-ojha** [comment]
jenkins test api

**Pull request merged**